echoregions/regions2d/regions2d.py

```diff
@@ -599,104 +653,79 @@ def transect_mask(
-        # If over a minute, throw an error.
-        max_time = (transect_df["region_bbox_right"] - transect_df["region_bbox_left"]).max()
-        max_time_minutes = max_time.total_seconds() / 60
-        if max_time_minutes > bbox_distance_threshold:
-            warning_messages.append(
-                f"Maximum width in time of transect log region bboxs is "
-                f"too large i.e. over {bbox_distance_threshold} minute(s). "
-                f"The maximum width is: {max_time_minutes}.",
-            )
-
-        # Raise an exception if there are any warning messages and must_pass_check is True.
-        # Else, print warning messages.
-        if len(warning_messages) > 0 and must_pass_check:
-            raise Exception("\n".join(warning_messages))
-        else:
-            print("\n".join(warning_messages))
+        # Check transect sequences
+        _check_transect_sequences(transect_df, must_pass_check)
 
         # Create binary variable indicating within transect segments.
         transect_df["within_transect"] = False
 
         # Indices where start_str followed by break_str/end_str
         st_indices = (transect_df["transect_type"] == start_str) & transect_df[
             "transect_type_next"
-        ].isin([break_str, end_str])
+        ].isin(transect_sequence_type_next_allowable_dict[start_str])
         transect_df.loc[st_indices, "within_transect"] = True
 
         # Indices where resume_str followed by break_str/end_str
         rt_indices = (transect_df["transect_type"] == resume_str) & transect_df[
             "transect_type_next"
-        ].isin([break_str, end_str])
+        ].isin(transect_sequence_type_next_allowable_dict[resume_str])
         transect_df.loc[rt_indices, "within_transect"] = True
 
-        # Get all unique file_names in transect_df.
-        transect_querying_list = list(transect_df["file_name"].unique())
-
-        # Create list of masks for each file name to be queried.
-        mask_list = []
-        for transect_querying_file_name in transect_querying_list:
-            within_transect_df = transect_df.query(
-                f'file_name == "{transect_querying_file_name}" and within_transect == True'
-            )
-            T = xr.zeros_like(da_Sv)
-            for _, row in within_transect_df.iterrows():
-                T = T + xr.where(
-                    (T.ping_time > row["region_bbox_left"])
-                    & (T.ping_time < row["region_bbox_left_next"]),
-                    1,
-                    0,
-                )
-            mask_list.append(T)
+        # Extract the min and max timestamps for filtering
+        min_timestamp = da_Sv.ping_time.min().values
+        max_timestamp = da_Sv.ping_time.max().values
+
+        # Find the index of the row right before min_timestamp
+        first_index_before_min = transect_df.loc[
+            transect_df["region_bbox_left"] < min_timestamp, "region_bbox_left"
+        ].idxmax()
```

Comment:
I think this should be "last" index before min?
```suggestion
        last_index_before_min = transect_df.loc[
            transect_df["region_bbox_left"] < min_timestamp, "region_bbox_left"
        ].idxmax()
```

Reply:
Oh yeah, I think that makes more sense. The directionality I was initially thinking of was more in reference to "if we look backwards in indices, what is the first index we hit", but yes, I can see why "last index" would be more appropriate here.